### PR TITLE
baremetal: Allow rootDeviceHints to override Host profiles

### PR DESCRIPTION
**Pull request by derekhiggins**
Hardware profiles don't allow for all of the use cases
required in some baremetal environments, allow the
boot hints associated with them to be overridden.
    
e.g. to specifiy a device by serial number
```
      - name: openshift-master-0
        rootDeviceHints:
          serialNumber: 1111

```

### CODE REVIEW

**Comment by stbenjam** [comment]
The go tools need appeasing, I think vendoring needs a go mod vendor/go mod tidy.  Other than that I like this.

@rlopez133 FYI - this has impact on your documentation/playbooks for 4.6.

File: docs/user/metal/install_ipi.md [hunk 1]
```diff
@@ -134,14 +134,20 @@ platform should be considered experimental and still subject to change without
...
+  The `hardwareProfile` is currently exposed as a way to allow specifying
   different hardware parameters for deployment.  By default, we will deploy
   RHCOS to the first disk, but that may not be appropriate for all hardware.
   The `hardwareProfile` is the field we have available to change that.  This
   interface is subject to change.  In the meantime, hardware profiles can be
   found here:
   https://github.com/metal3-io/baremetal-operator/blob/master/pkg/hardware/profile.go#L48
 
+* *rootDeviceHints* -- Guidance for how to choose the device to
+  receive the image being provisioned. Documentationon using this field can
```
Comment (stbenjam) [inline docs/user/metal/install_ipi.md:147]:
```suggestion
  receive the image being provisioned. Documentation using this field can
```

**Comment by stbenjam** [comment]
gofmt still needs appeasing too

**Comment by stbenjam** [comment]
/lgtm
/assign @sdodson 

This touches vendoring.

**Comment by sdodson** [comment]
/hold
to stop the looping, clear this whenever you get in

**Comment by derekhiggins** [comment]
/unhold
looks like e2e-aws is now working

**Comment by rlopez133** [comment]
@stbenjam  reading the install_ipi.md changes on this PR regarding rootDeviceHints which then points you here: https://github.com/metal3-io/baremetal-operator/blob/master/docs/api.md but I don't see any example on how to use it (unless I missed it). Can an example of rootDeviceHints being used within an install-config.yaml be included? Thanks!

**Comment by derekhiggins** [comment]
> @stbenjam reading the install_ipi.md changes on this PR regarding rootDeviceHints which then points you here: https://github.com/metal3-io/baremetal-operator/blob/master/docs/api.md but I don't see any example on how to use it (unless I missed it). Can an example of rootDeviceHints being used within an install-config.yaml be included? Thanks!

I didn't see this comment in time, I've submitted an example here https://github.com/openshift/installer/pull/3843

**Comment by stbenjam** [comment]
/retitle Bug 1805237: baremetal: Allow rootDeviceHints to override Host profiles

**Pull request merged**